src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -246,6 +246,11 @@ void java_bytecode_convert_classt::convert(
 /// java::array[X] where X is byte, float, int, char...
 void java_bytecode_convert_classt::add_array_types()
 {
+  // this method does not need to be called more than once
+  static bool already_executed=false;
+  if(already_executed)
+    return;
+
```

Comment:
To be honest, this sounds even worse. The place this belongs to is the point where it is being called. Apparently there can only be a single such call, as otherwise you'd be violating these invariants. That code seemingly knows that it has done the call, hence has all the necessarily state.